File: frontend/src/services/playerService.js
import api from './api'

const playerService = {
  getAllPlayers() {
    return api.get('/players')
  },

  getPlayerById(id) {
    return api.get(`/players/${id}`)
  },

  createPlayer(playerData) {
    return api.post('/players', playerData)
  },

  updatePlayer(id, playerData) {
    return api.put(`/players/${id}`, playerData)
  },

  deletePlayer(id) {
    return api.delete(`/players/${id}`)
  },

  getPlayersByTeam(teamId) {
    return api.get(`/players/team/${teamId}`)
  },

  getPlayersBySeason(seasonId) {
    return api.get(`/players/season/${seasonId}`)
  },

  getPlayersByTeamAndSeason(teamId, seasonId) {
    return api.get(`/players/team/${teamId}/season/${seasonId}`)
  }
}

export default playerService

File: backend/src/main/java/com/football/service/PlayerService.java
package com.football.service;

import com.football.dto.PlayerDTO;
import com.football.model.entity.Player;

import java.util.List;

public interface PlayerService {
    List<PlayerDTO> getAllPlayers();
    PlayerDTO getPlayerById(Integer playerId);
    PlayerDTO createPlayer(PlayerDTO playerDTO);
    PlayerDTO updatePlayer(Integer playerId, PlayerDTO playerDTO);
    void deletePlayer(Integer playerId);
    List<PlayerDTO> getPlayersByTeam(Integer teamId);
    List<PlayerDTO> getPlayersBySeason(Integer seasonId);
    List<PlayerDTO> getPlayersByTeamAndSeason(Integer teamId, Integer seasonId);
}
